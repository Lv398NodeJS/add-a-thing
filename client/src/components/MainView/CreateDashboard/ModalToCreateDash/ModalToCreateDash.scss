.modal-form {
  width: 90%;
  margin-left: 5%;
  margin-right: 5%;
  margin-top: 10;
}
