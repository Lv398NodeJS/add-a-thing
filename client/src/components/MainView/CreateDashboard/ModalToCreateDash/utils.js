export default (dashName, dashDescription, addDashboard, userId) => {
  const dashboard = {
    name: dashName,
    description: dashDescription,
    userId: userId,
  };
  addDashboard(dashboard);
};
