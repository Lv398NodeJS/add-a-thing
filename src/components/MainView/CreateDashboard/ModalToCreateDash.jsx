import React, { Component } from 'react';
import { Modal, Form, Button } from 'react-bootstrap';
import db from '../../../fire';
import createNewDash from './PushDashboardtoDb';
import './ModalToCreateDash.scss';

class ModalToCreateDash extends Component {
  constructor() {
    super();
    this.state = {
      show: true,
      dashName: '',
      dashDescription: '',
    };
  }

  handleSaveButtonPush = () => {
    const { addDashboard } = this.props;
    const { dashName, dashDescription } = this.state;
    createNewDash(dashName, dashDescription, addDashboard);
    this.setState({
      show: false,
      dashName: '',
      dashDescription: '',
    });
  }

  handleClose = () => {
    this.setState({
      show: false,
    });
  }

  handleSave = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  }

  render() {
    const { show, dashName, dashDescription } = this.state;
    return (
      <Modal
        size="lg"
        show={show}
        onHide={this.handleClose}
        className="createBtnModal"
      >
        <Modal.Header closeButton>
          <Modal.Title>Create Dashboard</Modal.Title>
        </Modal.Header>
        <Form.Control
          className="modal-form"
          type="text"
          placeholder="Enter the name for the new dashboard"
          name="dashName"
          value={dashName}
          onChange={this.handleSave}
        />
        <Form.Group controlId="exampleForm.ControlTextarea1">
          <Form.Control
            className="modal-form"
            as="textarea"
            placeholder="Enter the description for the new dashboard"
            name="dashDescription"
            value={dashDescription}
            onChange={this.handleSave}
          />
        </Form.Group>
        <br />
        <Modal.Body>Click &apos;Save Changes&apos; to create a new dashboard</Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={this.handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={this.handleSaveButtonPush} disabled={!dashName}>
            Save Changes
          </Button>
        </Modal.Footer>
      </Modal>
    );
  }
}

export default ModalToCreateDash;
