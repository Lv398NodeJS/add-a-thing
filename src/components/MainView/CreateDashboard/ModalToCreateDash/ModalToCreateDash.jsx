import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Modal, Form, Button } from 'react-bootstrap';
import createNewDash from './CreateNewDash';
import './ModalToCreateDash.scss';
import * as viewActions from '../../../../actions/mainViewActions';

export class ModalToCreateDash extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: props.show,
      dashName: '',
      dashDescription: '',
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ show: nextProps.show });
  }

  handleClose = () => {
    const { closeModal } = this.props;
    closeModal();
    this.setState({
      show: false,
      dashName: '',
      dashDescription: '',
    });
  }

  handleSaveButtonPush = () => {
    const { closeModal, mainViewActions } = this.props;
    const { dashName, dashDescription } = this.state;
    createNewDash(dashName, dashDescription, mainViewActions.addDashboard);
    closeModal();
    this.setState({
      show: false,
      dashName: '',
      dashDescription: '',
    });
  }

  handleSave = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  }

  render() {
    const { dashName, dashDescription, show } = this.state;
    return (
      <Modal
        onHide={this.handleClose}
        show={show}
        size="lg"
        className="createBtnModal"
      >
        <Modal.Header closeButton>
          <Modal.Title>Create Dashboard</Modal.Title>
        </Modal.Header>
        <Form.Control
          className="modal-form dash-name"
          type="text"
          placeholder="Enter the name for the new dashboard"
          name="dashName"
          value={dashName}
          onChange={this.handleSave}
        />
        <Form.Group controlId="exampleForm.ControlTextarea1">
          <Form.Control
            className="modal-form dash-description"
            as="textarea"
            placeholder="Enter the description for the new dashboard"
            name="dashDescription"
            value={dashDescription}
            onChange={this.handleSave}
          />
        </Form.Group>
        <br />
        <Modal.Body>Click &apos;Save Changes&apos; to create a new dashboard</Modal.Body>
        <Modal.Footer>
          <Button
            className="close-button"
            variant="secondary"
            onClick={this.handleClose}
          >
            Close
          </Button>
          <Button
            className="save-changes"
            variant="primary"
            onClick={this.handleSaveButtonPush}
            disabled={!dashName}
          >
            Save Changes
          </Button>
        </Modal.Footer>
      </Modal>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  mainViewActions: bindActionCreators(viewActions, dispatch),
});

export default connect(
  null,
  mapDispatchToProps,
)(ModalToCreateDash);
