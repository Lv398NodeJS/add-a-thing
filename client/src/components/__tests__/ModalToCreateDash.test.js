import React from 'react';
import { shallow } from 'enzyme';
import { ModalToCreateDash } from '../MainView/CreateDashboard/ModalToCreateDash/ModalToCreateDash';

describe('ModalToCreateDash', () => {
  const component = shallow(
    <ModalToCreateDash />,
  );

  it('should render correctly in "debug" mode', () => {
    const componentDebug = shallow(
      <ModalToCreateDash debug />,
    );

    expect(componentDebug).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    expect(component).toMatchSnapshot();
  });

  it('input fields should be filled correctly', () => {
    const dashInfo = { dashName: 'The awesome dashboard', dashDescriprtion: 'How awesome am I?' };
    expect(component.find('.dash-name').length).toBe(1);

    const nameInput = component.find('.dash-name');
    nameInput.value = dashInfo.dashName;
    expect(nameInput.value).toBe('The awesome dashboard');

    const descInput = component.find('.dash-description');
    descInput.value = dashInfo.dashDescriprtion;
    expect(descInput.value).toBe('How awesome am I?');
  });

  it("'Save changes' button should be disabled if dash name input field is empty", () => {
    expect(component.find('.save-changes').props().disabled).toBe(true);
  });

  it("'Save changes' button should be enabled when dash name input field is not empty", () => {
    component.setState({ dashName: 'I am not empty!' });

    expect(component.find('.save-changes').props().disabled).toBe(false);
  });

  it('closes modal window when close button clicked', () => {
    const shallowWithProps = shallow(
      <ModalToCreateDash closeModal={jest.fn()} />,
    );

    shallowWithProps.setState({ show: false });
    shallowWithProps.find('.close-button').simulate('click');

    expect(shallowWithProps.state().show).toEqual(false);
  });

  it("closes when 'Save changes' button clicked", () => {
    const shallowWithProps = shallow(
      <ModalToCreateDash
        closeModal={jest.fn()}
        mainViewActions={{ addDashboard: jest.fn() }}
      />,
    );

    shallowWithProps.setState({
      show: true,
      dashName: 'I am not empty!',
    });
    shallowWithProps.find('.save-changes').simulate('click');

    expect(shallowWithProps.state().show).toEqual(false);
  });
});
