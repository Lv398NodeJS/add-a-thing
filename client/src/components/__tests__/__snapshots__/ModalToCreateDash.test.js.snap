// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalToCreateDash should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
